tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -150,24 +150,24 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
   private def emitModuleImports(orderedClasses: List[LinkedClass],
       builder: JSTreeBuilder, logger: Logger): Unit = {
 
-    def foreachImportedModule(f: (String, Position) => Unit): Unit = {
+    def foreachImportedModule(f: (String, Position,  List[String]) => Unit): Unit = {
       val encounteredModuleNames = mutable.Set.empty[String]
       for (classDef <- orderedClasses) {
-        def addModuleRef(module: String): Unit = {
+        def addModuleRef(module: String, path: List[String]): Unit = {
           if (encounteredModuleNames.add(module))
-            f(module, classDef.pos)
+            f(module, classDef.pos, path)
```

Comment:
This will be problematic if there are two imports from the same module with different paths in the program. The first import will override the others, and you will miss some paths in the generated imports.

For example:
```scala
@js.native
@JSImport("url", "URL")
class URL(input: String) extends js.Object

@js.native
@JSImport("url", "URLSearchParams")
class URLSearchParams extends js.Object
```
There is already a test with two different paths: one for `Namespace` and one for `Default`. It should have failed the test suite, but the failure was hidden by the fact that you lump namespace and default together a bit further down (which is not correct in general either).

I suggest you add specific tests with different paths that are not namespace nor default. For example the `URL` and `URLSearchParams` above.